content/docs/visualization/template_reference.md

```diff
@@ -0,0 +1,107 @@
+---
+title: Template reference
+sort_rank: 6
+---
+
+# Template reference
+
+Prometheus has support for templating in both the summary/description field of
+alerts and also exposed as html endpoints. Templates have the ability to run
+queries against the databases, iterate over data, use conditionals, format data
+etc. This is based on the [go templating](http://golang.org/pkg/text/template/)
+system.
+
+## Data Structures
+
+The primary structure is the sample, defined as:
+
+```
+type sample struct {
+        Labels map[string]string
+        Value  float64
+}
+```
+
+The `__name__` label is the name of the metric.
+
+`[]sample` means a list of samples.
+
+## Functions
+In addition to the [default
+functions](http://golang.org/pkg/text/template/#hdr-Functions) provided by go
+templating, prometheus provides functions to make writing templates easier.
+
+If functions are used in a pipeline, the pipeline value is passed as the last argument. 
+
+### Queries
+
+| Name          | Arguments     | Returns  | Notes    |
+| ------------- | ------------- | -------- | -------- |
+| query         | query string  | []sample | Queries the databases, does not support returning matrixes.  |
+| first         | []sample      | sample   | Equivalent to `index a 0`  |
+| label         | label, sample | string   | Equivalent to `index sample.Labels label`  |
+| value         | sample        | number   | Equivalent to `sample.Value`  |
+| sortByLabel   | label, []samples | []sample | Sorts the samples by the given label. Is stable.  |
+
+`first`, `label` and `value` are intended to make query results easily usable in pipelines.
+
+### Numbers
+
+| Name          | Arguments     | Returns |  Notes    |
+| ------------- | --------------| --------| --------- |
+| humanize      | number        | string  | Converts a number to a more readable format, using Metric Prefixes 
+| humanize1024  | number        | string  | Like `humanize`, but uses 1024 as the base rather than 1000 |
+| humanizeDuration | number        | string  | Converts a duration in seconds to a more readable format. |
+
+Humanizing functions are intended to produce reasonable output for consumption by humans, and are not guaranteed to return the same results between versions.
```

Comment:
between Prometheus versions, yes? Maybe clarify that by saying "between Prometheus versions"